arroyo/processing/strategies/produce.py

```diff
@@ -0,0 +1,86 @@
+import logging
+import time
+from collections import deque
+from concurrent.futures import Future
+from typing import Deque, Optional, Tuple
+
+from arroyo.backends.abstract import Producer
+from arroyo.processing.strategies.abstract import MessageRejected, ProcessingStrategy
+from arroyo.types import Commit, Message, Position, Topic, TPayload
+
+logger = logging.getLogger(__name__)
+
+
+class ProduceAndCommit(ProcessingStrategy[TPayload]):
+    """
+    For each message received, produces a message to the topic being passed in.
+    Waits on the futures and only once completed, commits the offset of the message.
+    """
+
+    def __init__(
+        self,
+        producer: Producer[TPayload],
+        topic: Topic,
+        commit: Commit,
+        max_buffer_size: int = 10000,
+    ):
+        self.__producer = producer
+        self.__topic = topic
+        self.__commit = commit
+        self.__max_buffer_size = max_buffer_size
+
+        self.__queue: Deque[
+            Tuple[Message[TPayload], Future[Message[TPayload]]]
+        ] = deque()
+
+        self.__closed = False
+
+    def poll(self) -> None:
+        while self.__queue:
+            message, future = self.__queue[0]
+
+            if not future.done():
+                break
+
+            self.__queue.popleft()
```

Comment:
I think this is missing the error cases. 
It is not checking the result of the future in order to propagate an exception if the production failed.
If the producer failed, the future will contain an exception

Reply:
Added it